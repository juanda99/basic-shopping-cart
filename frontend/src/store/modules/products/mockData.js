export const productData = [
  {
    id: 1,
    name: 'Soup',
    customerPrice: 199,
    cost: 186,
    inCart: false,
    quantity: 1,
  },
  {
    id: 2,
    name: 'Bread',
    customerPrice: 87,
    cost: 21,
    inCart: false,
    quantity: 1,
  },
  {
    id: 3,
    name: 'Cheese',
    customerPrice: 275,
    cost: 234,
    inCart: false,
    quantity: 1,
  },
  {
    id: 4,
    name: 'Milk',
    customerPrice: 67,
    cost: 61,
    inCart: false,
    quantity: 1,
  },
]

export const serverData = [
  {
    id: 1,
    name: 'Soup',
    customerPrice: 199,
    cost: 186,
  },
  {
    id: 2,
    name: 'Bread',
    customerPrice: 87,
    cost: 21,
  },
  {
    id: 3,
    name: 'Cheese',
    customerPrice: 275,
    cost: 234,
  },
  {
    id: 4,
    name: 'Milk',
    customerPrice: 67,
    cost: 61,
  },
]
export default productData
