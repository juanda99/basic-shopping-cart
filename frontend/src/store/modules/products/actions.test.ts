import {
  getProductsFailureAction,
  getProductsRequestAction,
  getProductsSuccessAction
} from "./actions";
import { serverData } from "./mockData";

import {productData} from './mockData'
describe('Products actions', () => {

  it('Get Products request action', () => {
    expect(getProductsRequestAction()).toMatchSnapshot();
  });

  it('Get Products  success action', () => {
    expect(getProductsSuccessAction(serverData)).toMatchSnapshot();
  });

  it('Get Products failure action', () => {
    expect(getProductsFailureAction()).toMatchSnapshot();
  });
});
