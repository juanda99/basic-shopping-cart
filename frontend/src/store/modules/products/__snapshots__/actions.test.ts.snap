// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products actions Get Products  success action 1`] = `
Object {
  "payload": Object {
    "data": Array [
      Object {
        "cost": 186,
        "customerPrice": 199,
        "id": 1,
        "name": "Soup",
      },
      Object {
        "cost": 21,
        "customerPrice": 87,
        "id": 2,
        "name": "Bread",
      },
      Object {
        "cost": 234,
        "customerPrice": 275,
        "id": 3,
        "name": "Cheese",
      },
      Object {
        "cost": 61,
        "customerPrice": 67,
        "id": 4,
        "name": "Milk",
      },
    ],
  },
  "type": "GET_PRODUCTS_SUCCESS",
}
`;

exports[`Products actions Get Products failure action 1`] = `
Object {
  "type": "GET_PRODUCTS_FAILURE",
}
`;

exports[`Products actions Get Products request action 1`] = `
Object {
  "type": "GET_PRODUCTS_REQUEST",
}
`;
